src/docker.rs

```diff
@@ -0,0 +1,139 @@
+use std::collections::HashSet;
+use std::iter::FromIterator;
+use std::str::Utf8Error;
+
+/// Extract ARG value from a Dockerfile content
+/// E.g
+/// ```dockerfile
+/// FROM node
+///
+/// ARG FOO
+/// ARG BAR=default
+/// ...
+/// ```
+///
+/// will return a vector of "foo" and "bar" stings
+pub fn extract_dockerfile_args(dockerfile_content: Vec<u8>) -> Result<HashSet<String>, Utf8Error> {
+    let lines = std::str::from_utf8(dockerfile_content.as_slice())?;
+    let lines = lines.lines();
+
+    let used_args = lines
+        .into_iter()
+        .filter(|line| line.to_uppercase().trim().starts_with("ARG"))
```

Comment:
Replace "ARG" by "ARG " (with a space at the end) to avoid matching "ARGENT" "ARGUMENT" ...
I would remove the trim also, as it is a command it cannot be indented out, so it would avoid matching unecessary stuff

Reply:
It makes sense. Thank you